plugins/logging/file

```diff
@@ -279,13 +279,14 @@ sub hook_logging {
     if (   !$self->{_f}
         || !$self->{_nosplit}
         || !$transaction
+        || !UNIVERSAL::can($transaction, 'notes')
         || !$transaction->notes('file-logged-this-session'))
     {
         unless (defined $self->maybe_reopen($transaction)) {
             return DECLINED;
         }
-        if (UNIVERSAL::can($transaction,'isa')) {
-            $transaction->notes('file-logged-this-session', 1) if $transaction;
+        if (UNIVERSAL::can($transaction,'isa') && $transaction && UNIVERSAL::can($transaction, 'notes')) {
```

Comment:
For that matter, is checking 'isa' redundant and perhaps should be dropped entirely?